src/spikeinterface/preprocessing/common_reference.py

```diff
@@ -1,25 +1,47 @@
+from __future__ import annotations
 import numpy as np
+from typing import Optional, Literal
 
 from spikeinterface.core.core_tools import define_function_from_class
 
 from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
 from ..core import get_closest_channels
+from spikeinterface.core.baserecording import BaseRecording
 
 from .filter import fix_dtype
 
 
 class CommonReferenceRecording(BasePreprocessor):
     """
-    Re-references the recording extractor traces.
+    Re-references the recording extractor traces. That is, the value of the traces are
+    shifted so the there is a new zero (reference).
+
+    The new reference can be estimated either by using common median reference (CMR) or
+    common average reference (CAR).
+
+    The new reference can be set three ways:
+        * "global": the median/average of all channels is set as the new reference.
+        That is, the median/average is subtracted from all channels.
+        * "single": In the simplest case, a single channel from the recorder is set as the new reference.
+        That is, the channel is subtracted from all other channels. To use this option, the `ref_channel_ids` argument
+        is used with a single channel id.
```

Comment:
```suggestion
        is used with a single channel id. Note that this option will zero out the reference channel.
```

Reply:
This is a good addition to make it clear. Thanks.